tests-foundry/common/BaseTest.t.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache2.0
+pragma solidity 0.8.4;
+
+import {console} from "forge-std/console.sol";
+
+import {System} from "../../contracts/System.sol";
+import {Test} from "forge-std/Test.sol";
+import {Deployer} from "../../scripts-foundry/Deployer.s.sol";
+
+abstract contract BaseTest is System, Test {
+    
+    uint constant internal MAX_ETH_VALUE = 10_000_000 ether; // or any other reasonable cap
+    uint constant internal ADDITIONAL_GAS_FEES = 100 ether; // or any other reasonable value
+
+    bool constant internal ACCEPT_PAYMENTS = true; // test EOAs/eth-accepting contracts
+    bool constant internal REJECT_PAYMENTS = false;  // test eth-rejecting (hostile?) contracts
+
+    Deployer internal s_deployer;
+    
+    bool internal s_acceptPayment; // mutable by design. to be set by a derived test on demand
+
+    constructor(bool accept) {
+        s_acceptPayment = accept;
+    }
+
+    receive() external payable {
+        if (!s_acceptPayment) { 
+            revert("payment rejected");
+        }
+    } 
+
+    function setUp() public virtual {
+        s_deployer = new Deployer();
```

Comment:
Invoking the deployer script to fetch the tested contract addresses, while not mandatory, is considered a bets practice. It allows you to test either locally deployed or testnet deployed contracts (resolved by the deployer) and also silently debugs the deployer itself as part of the test runs